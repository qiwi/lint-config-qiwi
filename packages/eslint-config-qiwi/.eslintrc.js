module.exports = require('../infra/.eslintrc.json')
