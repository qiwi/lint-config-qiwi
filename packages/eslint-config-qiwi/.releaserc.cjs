module.exports = {
  ...require('../../.releaserc.cjs'),
  changelog: 'changelog',
  ghPages: {
    branch: 'gh-pages',
    from: 'docs',
    to: 'eslint-config-qiwi'
  }
}