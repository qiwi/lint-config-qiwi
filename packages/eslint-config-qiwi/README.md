# eslint-config-qiwi

[![CI](https://github.com/qiwi/lint-config-qiwi/workflows/CI/badge.svg)](https://github.com/qiwi/lint-config-qiwi/actions)
[![Test Coverage](https://api.codeclimate.com/v1/badges/8738098b0f0f4825da8b/test_coverage)](https://codeclimate.com/github/qiwi/lint-config-qiwi/test_coverage)
[![Maintainability](https://api.codeclimate.com/v1/badges/8738098b0f0f4825da8b/maintainability)](https://codeclimate.com/github/qiwi/lint-config-qiwi/maintainability)
[![npm (scoped)](https://img.shields.io/npm/v/eslint-config-qiwi?label=npm&color=39f)](https://www.npmjs.com/package/eslint-config-qiwi)

[ESLint](https://eslint.org) rules related to QIWI projects.

## Install

```bash
npm add eslint-config-qiwi -D
yarn add eslint-config-qiwi -D
```

## Usage

Refer to `eslint-config-qiwi` in your [.eslintrc.json](https://eslint.org/docs/user-guide/configuring):

```json
{
  "extends": "eslint-config-qiwi"
}
```

[Run](https://eslint.org/docs/user-guide/command-line-interface) linting your TS code:

```bash
eslint -c /any/path/to/.eslintrc.json --fix
```

## Contribution

If you find this config not strong enough you may add additional rules, write some tests, and trigger assets generation.

```bash
yarn test:u
```

## License

MIT
