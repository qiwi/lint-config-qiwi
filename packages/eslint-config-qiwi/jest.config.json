{
  "preset": "ts-jest",
  "collectCoverage": true,
  "coveragePathIgnorePatterns": [
  ],
  "collectCoverageFrom": [
    "<rootDir>/src/main/**/*.(j|t)s"
  ],
  "testMatch": [
    "<rootDir>/src/test/ts/*.ts",
    "<rootDir>/src/test/js/*.js"
  ],
  "testPathIgnorePatterns": [
    "/node_modules/",
    "<rootDir>/src/test/rules"
  ],
  "moduleFileExtensions": [
    "ts",
    "tsx",
    "js",
    "jsx",
    "json",
    "node"
  ]
}
