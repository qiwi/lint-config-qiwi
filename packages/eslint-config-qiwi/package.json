{
  "name": "eslint-config-qiwi",
  "version": "1.13.2",
  "description": "ESLint rules related to QIWI projects",
  "main": "src/main/js/eslint-config-qiwi.js",
  "scripts": {
    "build": "yarn docs",
    "test": "yarn test:spec",
    "test:gen": "yarn jest --updateSnapshot",
    "test:spec": "jest -w=1 --config=jest.config.json",
    "test:report": "yarn test && yarn coveralls:push",
    "lint": "eslint --config src/main/js/eslint-config-qiwi.js src/main/js/eslint-config-qiwi.js",
    "docs": "typedoc --options ./typedoc.json ./src/main"
  },
  "files": [
    "README.md",
    "CHANGELOG.md",
    "src/main"
  ],
  "repository": {
    "type": "git",
    "url": "git+https://github.com/qiwi/lint-config-qiwi.git"
  },
  "keywords": [
    "eslint-config-qiwi"
  ],
  "bugs": {
    "url": "https://github.com/qiwi/lint-config-qiwi/issues"
  },
  "homepage": "https://github.com/qiwi/lint-config-qiwi#readme",
  "author": "QIWI <opensource@qiwi.com>",
  "license": "MIT",
  "dependencies": {
    "@typescript-eslint/eslint-plugin": "^4.28.2",
    "@typescript-eslint/parser": "^4.28.2",
    "eslint-config-standard-qiwi": "^16.0.3",
    "eslint-plugin-array-func": "^3.1.7",
    "eslint-plugin-import": "^2.23.4",
    "eslint-plugin-node": "^11.1.0",
    "eslint-plugin-optimize-regex": "^1.2.0",
    "eslint-plugin-promise": "^5.1.0",
    "eslint-plugin-react": "^7.24.0",
    "eslint-plugin-react-hooks": "^4.2.0",
    "eslint-plugin-simple-import-sort": "^7.0.0",
    "eslint-plugin-sonarjs": "^0.9.1",
    "eslint-plugin-unicorn": "^34.0.0",
    "typescript": "*"
  },
  "devDependencies": {
    "@swissquote/crafty-preset-jest": "^1.15.0",
    "@types/jest": "^26.0.24",
    "coveralls": "^3.1.1",
    "eslint": "^7.30.0",
    "jest": "^27.0.6",
    "stdstream-snapshot": "^1.2.5",
    "ts-jest": "^27.0.3"
  },
  "peerDependencies": {},
  "optionalDependencies": {}
}
