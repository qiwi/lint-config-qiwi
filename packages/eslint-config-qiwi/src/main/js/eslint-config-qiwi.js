module.exports = {
  env: {
    node: true,
    browser: true,
    es6: true,
    jest: true,
  },
  parser: '@typescript-eslint/parser',
  extends: [
    'eslint:recommended',
    'standard',
    'plugin:@typescript-eslint/eslint-recommended',
    'plugin:@typescript-eslint/recommended',
    'plugin:react/recommended',
    'plugin:sonarjs/recommended',
    'plugin:array-func/recommended',
  ],
  plugins: [
    'react',
    '@typescript-eslint',
    '@typescript-eslint/eslint-plugin',
    'react-hooks',
    'optimize-regex',
    'sonarjs',
    'unicorn',
  ],
  rules: {
    'eslint/no-unused-vars': 0,
    'optimize-regex/optimize-regex': 'warn',
    'react/prop-types': 0,
    'react/display-name': 0,
    'unicorn/consistent-function-scoping': 'error',
    'unicorn/custom-error-definition': 'off',
    'unicorn/error-message': 'error',
    'unicorn/escape-case': 'error',
    'unicorn/explicit-length-check': 'error',
    'unicorn/import-index': 'error',
    'unicorn/new-for-builtins': 'error',
    'unicorn/no-array-instanceof': 'error',
    'unicorn/no-console-spaces': 'error',
    'unicorn/no-fn-reference-in-iterator': 'off',
    'unicorn/no-for-loop': 'error',
    'unicorn/no-hex-escape': 'error',
    'unicorn/no-keyword-prefix': 'off',
    'no-nested-ternary': 'off',
    'unicorn/no-new-buffer': 'error',
    'unicorn/no-process-exit': 'error',
    'unicorn/no-unreadable-array-destructuring': 'error',
    'unicorn/no-unsafe-regex': 'off',
    'unicorn/no-unused-properties': 'off',
    'unicorn/no-zero-fractions': 'error',
    'unicorn/number-literal-case': 'error',
    'unicorn/prefer-add-event-listener': 'error',
    'unicorn/prefer-dataset': 'error',
    'unicorn/prefer-event-key': 'error',
    'unicorn/prefer-exponentiation-operator': 'error',
    'unicorn/prefer-flat-map': 'error',
    'unicorn/prefer-includes': 'error',
    'unicorn/prefer-node-append': 'error',
    'unicorn/prefer-node-remove': 'error',
    'unicorn/prefer-query-selector': 'error',
    'unicorn/prefer-reflect-apply': 'error',
    'unicorn/prefer-spread': 'error',
    'unicorn/prefer-starts-ends-with': 'error',
    'unicorn/prefer-text-content': 'error',
    'unicorn/prefer-type-error': 'error',
    'unicorn/regex-shorthand': 'error',
    'unicorn/throw-new-error': 'error',
    'array-func/prefer-array-from': 0,
    'sonarjs/cognitive-complexity': 0,
    '@typescript-eslint/no-unused-vars': 'error',
    '@typescript-eslint/interface-name-prefix': 'off',
    'no-useless-constructor': 'off',
    '@typescript-eslint/camelcase': 'off',
    '@typescript-eslint/no-explicit-any': 'off',
    '@typescript-eslint/explicit-function-return-type': 'off',
    '@typescript-eslint/no-use-before-define': 'off',
    '@typescript-eslint/member-delimiter-style': 'off',
    '@typescript-eslint/ban-ts-ignore': 'off',
    'no-return-await': 'error',
    '@typescript-eslint/ban-ts-comment': 0,
    'object-curly-spacing': [2, 'never'],
    'space-before-function-paren': ['error', {
      anonymous: 'never',
      named: 'never',
      asyncArrow: 'always',
    }],
    'comma-dangle': [
      'error', {
        arrays: 'always-multiline',
        exports: 'always-multiline',
        functions: 'never',
        imports: 'always-multiline',
        objects: 'always-multiline',
      },
    ],
  },
  globals: {
    Cypress: 'readonly',
    cy: 'readonly',
    it: 'readonly',
    expect: 'readonly',
    describe: 'readonly',
    fetch: 'readonly',
  },
  settings: {
    react: {
      version: 'latest',
    },
  },
  parserOptions: {
    ecmaFeatures: {
      jsx: true,
    },
    ecmaVersion: 9,
    sourceType: 'module',
  },
}
