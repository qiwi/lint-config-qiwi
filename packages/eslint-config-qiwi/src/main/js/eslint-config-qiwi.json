{
  "env": {
    "es2022": true,
    "browser": true,
    "node": true,
    "jest": true
  },
  "settings": {
    "react": {
      "version": "18"
    }
  },
  "plugins": [
    "react",
    "react-hooks",
    "sonarjs",
    "unicorn",
    "@typescript-eslint"
  ],
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "ecmaFeatures": {
      "jsx": true
    },
    "jsxPragma": null,
    "ecmaVersion": "latest",
    "sourceType": "module"
  },
  "rules": {
    "@typescript-eslint/adjacent-overload-signatures": "error",
    "standard/array-bracket-even-spacing": "off",
    "array-bracket-newline": "off",
    "vue/array-bracket-newline": "off",
    "array-bracket-spacing": "off",
    "vue/array-bracket-spacing": "off",
    "array-element-newline": "off",
    "arrow-parens": "off",
    "arrow-spacing": "off",
    "vue/arrow-spacing": "off",
    "@typescript-eslint/ban-ts-comment": "error",
    "@typescript-eslint/ban-types": "error",
    "unicorn/better-regex": "error",
    "block-spacing": "off",
    "vue/block-spacing": "off",
    "vue/block-tag-newline": "off",
    "flowtype/boolean-style": "off",
    "brace-style": "off",
    "@typescript-eslint/brace-style": "off",
    "vue/brace-style": "off",
    "unicorn/catch-error-name": "off",
    "sonarjs/cognitive-complexity": "error",
    "comma-dangle": "off",
    "@typescript-eslint/comma-dangle": "off",
    "vue/comma-dangle": "off",
    "comma-spacing": "off",
    "@typescript-eslint/comma-spacing": "off",
    "vue/comma-spacing": "off",
    "comma-style": "off",
    "vue/comma-style": "off",
    "standard/computed-property-even-spacing": "off",
    "computed-property-spacing": "off",
    "unicorn/consistent-destructuring": "error",
    "unicorn/consistent-function-scoping": "error",
    "constructor-super": "error",
    "curly": "off",
    "unicorn/custom-error-definition": "off",
    "flowtype/delimiter-dangle": "off",
    "react/display-name": "error",
    "dot-location": "off",
    "vue/dot-location": "off",
    "sonarjs/elseif-without-else": "off",
    "unicorn/empty-brace-spaces": "off",
    "eol-last": "off",
    "unicorn/error-message": "error",
    "unicorn/escape-case": "error",
    "react-hooks/exhaustive-deps": "warn",
    "unicorn/expiring-todo-comments": "error",
    "unicorn/explicit-length-check": "error",
    "unicorn/filename-case": "off",
    "for-direction": "error",
    "func-call-spacing": "off",
    "@typescript-eslint/func-call-spacing": "off",
    "vue/func-call-spacing": "off",
    "function-call-argument-newline": "off",
    "function-paren-newline": "off",
    "generator-star": "off",
    "generator-star-spacing": "off",
    "flowtype/generic-spacing": "off",
    "getter-return": "error",
    "vue/html-closing-bracket-newline": "off",
    "vue/html-closing-bracket-spacing": "off",
    "vue/html-end-tags": "off",
    "vue/html-indent": "off",
    "vue/html-quotes": "off",
    "vue/html-self-closing": "off",
    "implicit-arrow-linebreak": "off",
    "unicorn/import-style": "error",
    "indent": "off",
    "@typescript-eslint/indent": "off",
    "indent-legacy": "off",
    "react/jsx-child-element-spacing": "off",
    "react/jsx-closing-bracket-location": "off",
    "react/jsx-closing-tag-location": "off",
    "react/jsx-curly-newline": "off",
    "react/jsx-curly-spacing": "off",
    "react/jsx-equals-spacing": "off",
    "react/jsx-first-prop-new-line": "off",
    "react/jsx-indent": "off",
    "react/jsx-indent-props": "off",
    "react/jsx-key": "error",
    "react/jsx-max-props-per-line": "off",
    "react/jsx-newline": "off",
    "react/jsx-no-comment-textnodes": "error",
    "react/jsx-no-duplicate-props": "error",
    "react/jsx-no-target-blank": "error",
    "react/jsx-no-undef": "error",
    "react/jsx-one-expression-per-line": "off",
    "react/jsx-props-no-multi-spaces": "off",
    "jsx-quotes": "off",
    "react/jsx-space-before-closing": "off",
    "react/jsx-tag-spacing": "off",
    "react/jsx-uses-react": "off",
    "react/jsx-uses-vars": "error",
    "react/jsx-wrap-multilines": "off",
    "key-spacing": "off",
    "vue/key-spacing": "off",
    "keyword-spacing": "off",
    "@typescript-eslint/keyword-spacing": "off",
    "vue/keyword-spacing": "off",
    "linebreak-style": "off",
    "lines-around-comment": "off",
    "vue/max-attributes-per-line": "off",
    "max-len": "off",
    "vue/max-len": "off",
    "sonarjs/max-switch-cases": "error",
    "@typescript-eslint/member-delimiter-style": "off",
    "vue/multiline-html-element-content-newline": "off",
    "multiline-ternary": "off",
    "vue/mustache-interpolation-spacing": "off",
    "unicorn/new-for-builtins": "error",
    "new-parens": "off",
    "newline-per-chained-call": "off",
    "unicorn/no-abusive-eslint-disable": "off",
    "sonarjs/no-all-duplicated-branches": "error",
    "unicorn/no-array-callback-reference": "error",
    "no-array-constructor": "off",
    "@typescript-eslint/no-array-constructor": "error",
    "unicorn/no-array-for-each": "off",
    "unicorn/no-array-method-this-argument": "error",
    "unicorn/no-array-push-push": "error",
    "unicorn/no-array-reduce": "off",
    "no-arrow-condition": "off",
    "no-async-promise-executor": "error",
    "unicorn/no-await-expression-member": "warn",
    "no-case-declarations": "error",
    "react/no-children-prop": "error",
    "no-class-assign": "error",
    "sonarjs/no-collapsible-if": "error",
    "sonarjs/no-collection-size-mischeck": "error",
    "no-comma-dangle": "off",
    "no-compare-neg-zero": "error",
    "no-cond-assign": "error",
    "no-confusing-arrow": "off",
    "unicorn/no-console-spaces": "error",
    "no-const-assign": "error",
    "no-constant-condition": "error",
    "no-control-regex": "error",
    "react/no-danger-with-children": "error",
    "no-debugger": "error",
    "no-delete-var": "error",
    "react/no-deprecated": "error",
    "react/no-direct-mutation-state": "error",
    "unicorn/no-document-cookie": "error",
    "no-dupe-args": "error",
    "no-dupe-class-members": "error",
    "no-dupe-else-if": "error",
    "no-dupe-keys": "error",
    "no-duplicate-case": "error",
    "sonarjs/no-duplicate-string": "error",
    "sonarjs/no-duplicated-branches": "error",
    "sonarjs/no-element-overwrite": "error",
    "no-empty": "error",
    "no-empty-character-class": "error",
    "sonarjs/no-empty-collection": "error",
    "unicorn/no-empty-file": "error",
    "no-empty-function": "off",
    "@typescript-eslint/no-empty-function": "error",
    "@typescript-eslint/no-empty-interface": "error",
    "no-empty-pattern": "error",
    "no-ex-assign": "error",
    "@typescript-eslint/no-explicit-any": "warn",
    "sonarjs/no-extra-arguments": "error",
    "no-extra-boolean-cast": "error",
    "@typescript-eslint/no-extra-non-null-assertion": "error",
    "no-extra-parens": "off",
    "@typescript-eslint/no-extra-parens": "off",
    "vue/no-extra-parens": "off",
    "no-extra-semi": "off",
    "@typescript-eslint/no-extra-semi": "off",
    "no-fallthrough": "error",
    "react/no-find-dom-node": "error",
    "no-floating-decimal": "off",
    "unicorn/no-for-loop": "error",
    "no-func-assign": "error",
    "no-global-assign": "error",
    "sonarjs/no-gratuitous-expressions": "error",
    "unicorn/no-hex-escape": "error",
    "sonarjs/no-identical-conditions": "error",
    "sonarjs/no-identical-expressions": "error",
    "sonarjs/no-identical-functions": "error",
    "sonarjs/no-ignored-return": "error",
    "no-import-assign": "error",
    "@typescript-eslint/no-inferrable-types": "error",
    "no-inner-declarations": "error",
    "unicorn/no-instanceof-array": "error",
    "no-invalid-regexp": "error",
    "unicorn/no-invalid-remove-event-listener": "error",
    "sonarjs/no-inverted-boolean-check": "off",
    "no-irregular-whitespace": "error",
    "react/no-is-mounted": "error",
    "unicorn/no-keyword-prefix": "off",
    "unicorn/no-lonely-if": "error",
    "no-loss-of-precision": "off",
    "@typescript-eslint/no-loss-of-precision": "error",
    "no-misleading-character-class": "error",
    "@typescript-eslint/no-misused-new": "error",
    "no-mixed-operators": "off",
    "no-mixed-spaces-and-tabs": "off",
    "no-multi-spaces": "off",
    "vue/no-multi-spaces": "off",
    "no-multiple-empty-lines": "off",
    "@typescript-eslint/no-namespace": "error",
    "no-negated-condition": "off",
    "unicorn/no-negated-condition": "error",
    "sonarjs/no-nested-switch": "error",
    "sonarjs/no-nested-template-literals": "error",
    "no-nested-ternary": "off",
    "unicorn/no-nested-ternary": "off",
    "unicorn/no-new-array": "error",
    "unicorn/no-new-buffer": "error",
    "no-new-symbol": "error",
    "@typescript-eslint/no-non-null-asserted-optional-chain": "error",
    "@typescript-eslint/no-non-null-assertion": "warn",
    "no-nonoctal-decimal-escape": "error",
    "unicorn/no-null": "off",
    "no-obj-calls": "error",
    "unicorn/no-object-as-default-parameter": "error",
    "no-octal": "error",
    "sonarjs/no-one-iteration-loop": "error",
    "unicorn/no-process-exit": "error",
    "no-prototype-builtins": "error",
    "no-redeclare": "error",
    "sonarjs/no-redundant-boolean": "error",
    "sonarjs/no-redundant-jump": "error",
    "no-regex-spaces": "error",
    "react/no-render-return-value": "error",
    "no-reserved-keys": "off",
    "sonarjs/no-same-line-conditional": "error",
    "no-self-assign": "error",
    "no-setter-return": "error",
    "no-shadow-restricted-names": "error",
    "sonarjs/no-small-switch": "error",
    "no-space-before-semi": "off",
    "no-spaced-func": "off",
    "vue/no-spaces-around-equal-signs-in-attribute": "off",
    "no-sparse-arrays": "error",
    "unicorn/no-static-only-class": "error",
    "react/no-string-refs": "error",
    "no-tabs": "off",
    "unicorn/no-thenable": "error",
    "@typescript-eslint/no-this-alias": "error",
    "unicorn/no-this-assignment": "error",
    "no-this-before-super": "error",
    "no-trailing-spaces": "off",
    "unicorn/no-typeof-undefined": "error",
    "no-undef": "error",
    "react/no-unescaped-entities": "error",
    "no-unexpected-multiline": "off",
    "react/no-unknown-property": "error",
    "unicorn/no-unnecessary-await": "error",
    "@typescript-eslint/no-unnecessary-type-constraint": "error",
    "no-unreachable": "error",
    "unicorn/no-unreadable-array-destructuring": "error",
    "unicorn/no-unreadable-iife": "error",
    "react/no-unsafe": "off",
    "no-unsafe-finally": "error",
    "no-unsafe-negation": "error",
    "no-unsafe-optional-chaining": "error",
    "unicorn/no-unsafe-regex": "off",
    "sonarjs/no-unused-collection": "error",
    "no-unused-labels": "error",
    "unicorn/no-unused-properties": "off",
    "no-unused-vars": "off",
    "@typescript-eslint/no-unused-vars": "warn",
    "sonarjs/no-use-of-empty-return-value": "error",
    "no-useless-backreference": "error",
    "no-useless-catch": "error",
    "sonarjs/no-useless-catch": "error",
    "no-useless-escape": "error",
    "unicorn/no-useless-fallback-in-spread": "error",
    "unicorn/no-useless-length-check": "error",
    "unicorn/no-useless-promise-resolve-reject": "error",
    "unicorn/no-useless-spread": "error",
    "unicorn/no-useless-switch-case": "error",
    "unicorn/no-useless-undefined": "error",
    "@typescript-eslint/no-var-requires": "error",
    "no-whitespace-before-property": "off",
    "no-with": "error",
    "no-wrap-func": "off",
    "unicorn/no-zero-fractions": "error",
    "sonarjs/non-existent-operator": "error",
    "nonblock-statement-body-position": "off",
    "unicorn/number-literal-case": "off",
    "unicorn/numeric-separators-style": "error",
    "standard/object-curly-even-spacing": "off",
    "object-curly-newline": "off",
    "vue/object-curly-newline": "off",
    "object-curly-spacing": "off",
    "@babel/object-curly-spacing": "off",
    "@typescript-eslint/object-curly-spacing": "off",
    "babel/object-curly-spacing": "off",
    "vue/object-curly-spacing": "off",
    "object-property-newline": "off",
    "vue/object-property-newline": "off",
    "flowtype/object-type-curly-spacing": "off",
    "flowtype/object-type-delimiter": "off",
    "one-var-declaration-per-line": "off",
    "operator-linebreak": "off",
    "vue/operator-linebreak": "off",
    "padded-blocks": "off",
    "unicorn/prefer-add-event-listener": "error",
    "unicorn/prefer-array-find": "error",
    "unicorn/prefer-array-flat": "error",
    "unicorn/prefer-array-flat-map": "error",
    "unicorn/prefer-array-index-of": "error",
    "unicorn/prefer-array-some": "error",
    "@typescript-eslint/prefer-as-const": "error",
    "unicorn/prefer-at": "off",
    "unicorn/prefer-code-point": "error",
    "unicorn/prefer-date-now": "error",
    "unicorn/prefer-default-parameters": "error",
    "unicorn/prefer-dom-node-append": "error",
    "unicorn/prefer-dom-node-dataset": "error",
    "unicorn/prefer-dom-node-remove": "error",
    "unicorn/prefer-dom-node-text-content": "error",
    "unicorn/prefer-event-target": "off",
    "unicorn/prefer-export-from": "error",
    "sonarjs/prefer-immediate-return": "error",
    "unicorn/prefer-includes": "error",
    "unicorn/prefer-json-parse-buffer": "off",
    "unicorn/prefer-keyboard-event-key": "error",
    "unicorn/prefer-logical-operator-over-ternary": "error",
    "unicorn/prefer-math-trunc": "off",
    "unicorn/prefer-modern-dom-apis": "error",
    "unicorn/prefer-modern-math-apis": "error",
    "unicorn/prefer-module": "off",
    "@typescript-eslint/prefer-namespace-keyword": "error",
    "unicorn/prefer-native-coercion-functions": "error",
    "unicorn/prefer-negative-index": "error",
    "unicorn/prefer-node-protocol": "error",
    "unicorn/prefer-number-properties": "error",
    "unicorn/prefer-object-from-entries": "error",
    "sonarjs/prefer-object-literal": "error",
    "unicorn/prefer-optional-catch-binding": "error",
    "unicorn/prefer-prototype-methods": "error",
    "unicorn/prefer-query-selector": "error",
    "unicorn/prefer-reflect-apply": "error",
    "unicorn/prefer-regexp-test": "error",
    "unicorn/prefer-set-has": "error",
    "unicorn/prefer-set-size": "error",
    "sonarjs/prefer-single-boolean-return": "error",
    "unicorn/prefer-spread": "error",
    "unicorn/prefer-string-replace-all": "off",
    "unicorn/prefer-string-slice": "error",
    "unicorn/prefer-string-starts-ends-with": "error",
    "unicorn/prefer-string-trim-start-end": "error",
    "unicorn/prefer-switch": "error",
    "unicorn/prefer-ternary": "error",
    "unicorn/prefer-top-level-await": "warn",
    "unicorn/prefer-type-error": "error",
    "sonarjs/prefer-while": "error",
    "unicorn/prevent-abbreviations": "off",
    "react/prop-types": "error",
    "quote-props": "off",
    "vue/quote-props": "off",
    "quotes": "off",
    "@typescript-eslint/quotes": "off",
    "babel/quotes": "off",
    "flowtype/quotes": "off",
    "react/react-in-jsx-scope": "off",
    "unicorn/relative-url-style": "error",
    "unicorn/require-array-join-separator": "error",
    "unicorn/require-number-to-fixed-digits-argument": "error",
    "unicorn/require-post-message-target-origin": "off",
    "react/require-render-return": "error",
    "require-yield": "error",
    "rest-spread-spacing": "off",
    "react-hooks/rules-of-hooks": "error",
    "vue/script-indent": "off",
    "semi": "off",
    "@babel/semi": "off",
    "@typescript-eslint/semi": "off",
    "babel/semi": "off",
    "flowtype/semi": "off",
    "semi-spacing": "off",
    "semi-style": "off",
    "vue/singleline-html-element-content-newline": "off",
    "space-after-function-name": "off",
    "space-after-keywords": "off",
    "flowtype/space-after-type-colon": "off",
    "space-before-blocks": "off",
    "@typescript-eslint/space-before-blocks": "off",
    "space-before-function-paren": "off",
    "@typescript-eslint/space-before-function-paren": "off",
    "space-before-function-parentheses": "off",
    "flowtype/space-before-generic-bracket": "off",
    "space-before-keywords": "off",
    "flowtype/space-before-type-colon": "off",
    "space-in-brackets": "off",
    "space-in-parens": "off",
    "vue/space-in-parens": "off",
    "space-infix-ops": "off",
    "@typescript-eslint/space-infix-ops": "off",
    "vue/space-infix-ops": "off",
    "space-return-throw-case": "off",
    "space-unary-ops": "off",
    "vue/space-unary-ops": "off",
    "space-unary-word-ops": "off",
    "unicorn/string-content": "off",
    "unicorn/switch-case-braces": "error",
    "switch-colon-spacing": "off",
    "template-curly-spacing": "off",
    "vue/template-curly-spacing": "off",
    "unicorn/template-indent": "error",
    "template-tag-spacing": "off",
    "unicorn/text-encoding-identifier-case": "off",
    "unicorn/throw-new-error": "error",
    "@typescript-eslint/triple-slash-reference": "error",
    "@typescript-eslint/type-annotation-spacing": "off",
    "unicode-bom": "off",
    "flowtype/union-intersection-spacing": "off",
    "use-isnan": "error",
    "valid-typeof": "error",
    "wrap-iife": "off",
    "wrap-regex": "off",
    "yield-star-spacing": "off"
  },
  "overrides": [
    {
      "files": [
        "*.ts",
        "*.tsx",
        "*.mts",
        "*.cts"
      ],
      "rules": {
        "constructor-super": "off",
        "getter-return": "off",
        "no-const-assign": "off",
        "no-dupe-args": "off",
        "no-dupe-class-members": "off",
        "no-dupe-keys": "off",
        "no-func-assign": "off",
        "no-import-assign": "off",
        "no-new-symbol": "off",
        "no-obj-calls": "off",
        "no-redeclare": "off",
        "no-setter-return": "off",
        "no-this-before-super": "off",
        "no-undef": "off",
        "no-unreachable": "off",
        "no-unsafe-negation": "off",
        "no-var": "error",
        "prefer-const": "error",
        "prefer-rest-params": "error",
        "prefer-spread": "error",
        "valid-typeof": "off"
      }
    }
  ]
}
