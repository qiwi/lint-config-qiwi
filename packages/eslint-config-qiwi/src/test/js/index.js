import {generateSnapshot} from 'stdstream-snapshot'
import {resolve} from 'path'
import * as config from '../../main/js/eslint-config-qiwi.js'

describe('eslint-config-qiwi (JavaScript)', () => {
  it('config is exposed as js', () => {
    expect(config).toEqual(expect.any(Object))
  })

  it('config works as expected', async () => {
    const configPath = resolve(__dirname, '../../main/js/eslint-config-qiwi.js')
    const rulesPath = resolve(__dirname, 'rules/*.js')
    const eslint = resolve(__dirname, '../../../../../node_modules/.bin/eslint')
    const result = await generateSnapshot({
      cmd: `${eslint} --config ${configPath} ${rulesPath}`
    })

    expect(result).toMatchSnapshot()
  })
})
