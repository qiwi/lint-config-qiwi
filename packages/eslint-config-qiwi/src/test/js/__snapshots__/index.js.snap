// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`eslint-config-qiwi (JavaScript) config works as expected 1`] = `
Object {
  "err": Object {
    "code": 1,
    "killed": false,
    "signal": null,
  },
  "stderr": "",
  "stdout": "/src/test/js/rules/trailing-comma-fail.js
  3:7   error  Missing trailing comma     comma-dangle
  9:10  error  Unexpected trailing comma  comma-dangle

âœ– 2 problems (2 errors, 0 warnings)
  2 errors and 0 warnings potentially fixable with the \`--fix\` option.",
}
`;
